### PR TITLE
Da 325/create asset with stats

### PR DESCRIPTION
**Pull request by polymath-eric**
### Description

Adds ability to create an Asset with statistics

### Breaking Changes

N/A

### JIRA Link

[DA-325](https://polymath.atlassian.net/browse/DA-325)

### CODE REVIEW

File: src/api/procedures/types.ts [hunk 1]
```diff
@@ -236,6 +236,18 @@ export interface CreateAssetParams {
...
+
+  /**
+   * (optional) type of statistics that should be enabled for the Asset
+   *
+   * Enabling statistics allows for TransferRestrictions to be made. For example the SEC requires registration for a company that
+   * has either more than 2000 investors, or more than 500 non accredited investors. To prevent crossing this limit two restrictions are
+   * needed, a `Count` of 2000, and a `ScopedCount` of non accredited with a maximum of 500. [source](https://www.sec.gov/info/smallbus/secg/jobs-act-section-12g-small-business-compliance-guide.htm)
+   *
+   * These restrictions require a `Count` and `ScopedCount` statistic to be created. Although they an be created after the Asset is made, it is recommended to create statistics
+   * before the Asset is circulated. Count statistics made after Asset creation need their initial value set, so it is simpler to create them before investors hold the Asset.
```
Comment (prashantasdeveloper) [inline src/api/procedures/types.ts:248]: nit-> We can also add link to `batchUpdateAssetStats` to make it clear how to set an asset stat value if it has not been created at the time of asset creation

**Review by prashantasdeveloper — APPROVED** [review]
LGTM

**Review by monitz87 — CHANGES_REQUESTED** [review]

File: src/api/procedures/types.ts (hunk 1 above)
Comment (polymath-eric) [reply src/api/procedures/types.ts:248]: Thats the raw chain method, which I don't think we should link from the SDK. I think linking to `enableStat` would work though, it takes in initial values as input when needed.

File: src/types/index.ts [hunk 2]
```diff
@@ -339,11 +339,47 @@ export type StatClaimInput =
...
+      type: StatType.Balance;
+    }
+  | {
+      type: StatType.ScopedCount;
+      claimIssuer: StatClaimIssuer;
+    }
+  | {
+      type: StatType.ScopedBalance;
+      claimIssuer: StatClaimIssuer;
+    };
```
[the remark this answers is not in the document]
Comment (polymath-eric) [reply src/types/index.ts]: I went with the union type. If there is a new stat type I think we'd need to add it in. If it happens to use claimIssuer like this, then it should be just as easy to add in.

File: src/utils/conversion.ts [hunk 3]
```diff
@@ -544,7 +544,7 @@ export function permillToBigNumber(value: Permill): BigNumber {
 /**
  *  @hidden
  */
-export function meshClaimToStatClaimInput(claim: StatClaim): StatClaimInput {
+export function meshClaimToStatClaimInput(claim: StatClaim): InputStatClaim {
```
Comment (monitz87) [inline src/utils/conversion.ts:547]: ToInputStatClaim

**Review by monitz87 — CHANGES_REQUESTED** [review]
Apparently something went wrong with a find and replace, there's a lot of lines that aren't valid typescript

**Review by monitz87 — APPROVED** [review]

**Pull request merged**